tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/ModifyRepositoryMetadataMojo.java

```diff
@@ -0,0 +1,159 @@
+ * subsequently back to the remote server.
+ * </p>
+ */
+@Mojo(name = "modify-metadata", threadSafe = true)
+public class ModifyRepositoryMetadataMojo extends AbstractP2Mojo {
+
+    /**
+     * Location of the <em>source</em> repository to modify.<br>
+     * May be the {@code Path} of a local file or {@code URL}/{@code URI} to a remote
+     * repository.<br>
+     * If this is set to a remote URL the {@link #outputLocation} must be set to a local path as
+     * remote locations cannot be written.
+     */
+    @Parameter(property = "p2.repository.location")
+    private URI repositoryLocation;
+
+    /**
+     * Optional output location where the modified metadata files ({@code artifact.xml/jar/xml.xz}
+     * or {@code content.xml/jar/xml.xz}) are written to.<br>
+     * If not specified or equals to {@link #repositoryLocation} the source repository is modified
+     * in-place.
+     */
+    @Parameter(property = "p2.repository.output")
+    private Path outputLocation;
+
+    /**
+     * The kind of modified repository: Either {@code artifact} or {@code metadata}<br>
+     * If {@code artifact} is specified, only the {@code artifact.xml/jar/xml.xz} files are
+     * modified, if {@code metadata} only {@code content.xml/jar/xml.xz}. If blank (the default)
+     * both are modified.
+     */
+    @Parameter(property = "p2.repository.kind", defaultValue = "")
+    private String repositoryKind;
+
+    /** Optional repository name to set. An existing name is overwritten. */
+    @Parameter
+    private String repositoryName;
+    /**
+     * The properties to remove.<br>
+     * Properties that don't exist in the repository are just silently ignored.
+     */
+    @Parameter
+    private List<String> propertiesToRemove = List.of();
+    /**
+     * The properties to add.<br>
+     * Already existing properties are overwritten and updated to the value specified.
+     */
+    @Parameter
+    private Map<String, String> propertiesToAdd = Map.of();
+
+    /**
+     * <p>
+     * Compress the repository index files <code>content.xml</code> and <code>artifacts.xml</code>.
+     * </p>
+     */
+    @Parameter(defaultValue = "true")
+    private boolean compress;
+
+    /**
+     * <p>
+     * Add XZ-compressed repository index files. XZ offers better compression ratios esp. for highly
+     * redundant file content.
+     * </p>
+     */
+    @Parameter(defaultValue = "true")
+    private boolean xzCompress;
+
+    /**
+     * <p>
+     * If {@link #xzCompress} is <code>true</code>, whether jar or xml index files should be kept in
+     * addition to XZ-compressed index files. This fallback provides backwards compatibility for
+     * pre-Mars p2 clients which cannot read XZ-compressed index files.
+     * </p>
+     */
+    @Parameter(defaultValue = "true")
+    private boolean keepNonXzIndexFiles;
+
+    @Inject
+    private RepositoryModifier modifier;
```

Comment:
I must confess the name looks odd here, a modifier is usually something that modifies the behavior (e.g. a flag), what about RepositoryUpdater?

Reply:
In this case it modifies the content of a P2 repository. It also does not really update a repository as it does general modifications, which might add something new but could also just remove something.

What about `RepositoryTransformer`, `RepositoryMutator` or `RepositoryProcessor`?
I'm currently leaning towards the first one. But in the end it's internal and can be changed at any time.

Reply:
I don't think this thing "manages" something, in any case we can later rename it if we find another name more suitable, this is nothing that faces the user visible API.